### ISSUE TITLE
warn with uplevel gives 'TypeError (wrong argument type Hash (expected String))' in ruby 2.5/2.6

### ISSUE THREAD

**Issue by rockorequin** [schmidt/structured_warnings#10]
Using the 0.3.0 structured warnings gem, when I run the following code in irb 2.5.3 or 2.6.2 using the big_decimal 1.4.3 gem:

```
gem 'structured_warnings', '=0.3.0'
require 'structured_warnings'
gem 'bigdecimal', '=1.4.3'
require 'bigdecimal'
BigDecimal.new 0
```
then I get this exception:

```
Traceback (most recent call last):
        8: from /usr/local/rvm/rubies/ruby-2.6.2/bin/irb:23:in `<main>'
        7: from /usr/local/rvm/rubies/ruby-2.6.2/bin/irb:23:in `load'
        6: from /usr/local/rvm/rubies/ruby-2.6.2/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        5: from (irb):3
        4: from /usr/local/rvm/gems/ruby-2.6.2/gems/bigdecimal-1.4.3/lib/bigdecimal.rb:10:in `new'
        3: from /usr/local/rvm/gems/ruby-2.6.2/gems/structured_warnings-0.3.0/lib/structured_warnings/kernel.rb:3:in `warn'
        2: from /usr/local/rvm/gems/ruby-2.6.2/gems/structured_warnings-0.3.0/lib/structured_warnings/warning.rb:62:in `warn'
        1: from /usr/local/rvm/gems/ruby-2.6.2/gems/structured_warnings-0.3.0/lib/structured_warnings/warning.rb:62:in `warn'
TypeError (wrong argument type Hash (expected String))
```

The exception seems to be raised because the warn method doesn't like the uplevel: 1 hash that BigDecimal.new sends in this line:

`warn "BigDecimal.new is deprecated; use BigDecimal() method instead.", uplevel: 1`

By inserting a puts statement before this code in kernel_warn.rb, I can see that it is being called:

module_function define_method(:warn) {|*messages, uplevel: nil|
...

but a puts inside the definition block never gets shown, so the redefinition doesn't seem to be working.

**Comment by schm** [comment]
Thanks for the bug report. I wasn't aware of the `uplevel` parameter for `warn`. It looks like it was added with Ruby 2.5.

I'll take a look at properly supporting the parameter soon.

**Comment by schmidt** [comment]
Thanks for the bug report. I wasn't aware of the uplevel parameter for warn. It looks like it was added with Ruby 2.5.

I'll take a look at properly supporting the parameter soon.

**Comment by rockorequin** [comment]
Thanks, that latest commit fixes the issue for me. (Although I get soooo many warnings now!)

**Comment by schmidt** [comment]
Released with 0.4.0
